socket_linux.go

```diff
@@ -10,7 +10,12 @@ import (
 
 // see also https://github.com/jbenet/go-reuseport/blob/master/impl_unix.go#L279
 func NewSocket(addr *net.UDPAddr, recvBuf int, reuseport bool) (net.PacketConn, error) {
-	sockFD, err := unix.Socket(unix.AF_INET, unix.SOCK_DGRAM|syscall.SOCK_CLOEXEC|syscall.SOCK_NONBLOCK, 0)
+	// default to AF_INET6 to be equivalent to net.ListenUDP()
+	domain := unix.AF_INET6
+	if len(addr.IP) == net.IPv4len {
```

Comment:
Should we use `addr.IP.To4() != nil` here instead? (It returns `nil` if it's not an IPv4 address and internally implements this same length check.)